### PR TITLE
drivers: amd: renoir: use sample rate from topology

### PR DESCRIPTION
**Pull request by balakishorepati**
So far we only support 48000Hz as default sample rate.
Now introduce acp_params structure to pass configuration
from topology via AP to DSP.

Only Sample rate param for now to configure dmic clk.

While here, align sof_ipc_dai_acp_params struct
with the one from kernel.

Increment only ABI MINOR version which makes the
SOF FW backward compatible with older kernel versions

Signed-off-by: Balakishorepati <balaKishore.pati@amd.com>

### CODE REVIEW

**Comment by balakishorepati** [comment]
Kernel PR for Same: https://github.com/thesofproject/linux/pull/3556

File: src/drivers/amd/renoir/acp_dmic_dai.c
Comment (ranj063) [inline src/drivers/amd/renoir/acp_dmic_dai.c:90]: I'd switch the set_drvdata to NULL and rfree

File: src/include/kernel/abi.h
Comment (ranj063) [inline src/include/kernel/abi.h:32]: What is this used for? How does this guarantee backward compatibility?

File: src/drivers/amd/renoir/acp_dmic_dai.c
Comment (balakishorepati) [reply src/drivers/amd/renoir/acp_dmic_dai.c:90]: Intel and other platforms also doing in same way.
For reference:
https://github.com/thesofproject/sof/blob/main/src/drivers/imx/esai.c#L372
https://github.com/thesofproject/sof/blob/main/src/drivers/intel/ssp/ssp.c#L1147

File: src/include/kernel/abi.h
Comment (balakishorepati) [reply src/include/kernel/abi.h:32]: This ABI MINOR version is required due to the changes in common dai structure b/w kernel and firmware. This will let you know  both kernel and Firmware should be aligned for use cases to work.
Also some of the Scripts were rejecting the patch for upstream if we don't update ABI version as the changes are updating the dai structures.

**Review by lyakh — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@balakishorepati is there a kernel PR to align ?
@plbossart fyi

**Review by sunilkumardommati — APPROVED** [review]

**Review by bhiregoudar — APPROVED** [review]

**Comment by balakishorepati** [comment]
> @balakishorepati best to use the ABI label to flag any IPC patches. I've added it.

**Comment by lgirdwood** [comment]
@balakishorepati think we are good to merge except we have an open on the ABI update MINOR value due to https://github.com/thesofproject/linux/pull/3353 which we need to align and unblock ABI.

**Comment by balakishorepati** [comment]
> @balakishorepati think we are good to merge except we have an open on the ABI update MINOR value due to [thesofproject/linux#3353](https://github.com/thesofproject/linux/pull/3353) which we need to align and unblock ABI.
> 
> Kernel PR : [thesofproject/linux#3556](https://github.com/thesofproject/linux/pull/3556) is already merged into sof main kernel.

@lgirdwood : Is there any thing pending from my end to merge this pull request ?

**Comment by lgirdwood** [comment]
> > @balakishorepati think we are good to merge except we have an open on the ABI update MINOR value due to [thesofproject/linux#3353](https://github.com/thesofproject/linux/pull/3353) which we need to align and unblock ABI.
> > Kernel PR : [thesofproject/linux#3556](https://github.com/thesofproject/linux/pull/3556) is already merged into sof main kernel.
> 
> @lgirdwood : Is there any thing pending from my end to merge this pull request ?

Just need to remove the merge commit from the PR.

**Comment by lgirdwood** [comment]
> > @lgirdwood : Is there any thing pending from my end to merge this pull request ?
> 
> Just need to remove the merge commit from the PR. @lgirdwood : Removed the merge commit. Can you plz merge the pull request.

Thanks, will merge as soon as we fix the kernel ABI update (all kernel and FW ABI PRs are blocked on this).

**Comment by sys-pt1s** [comment]
Can one of the admins verify this patch?

**Review by kv2019i — APPROVED** [review]

**Comment by lgirdwood** [comment]
rerun CI due to recent CI downtime.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@wszypelt this PR should have no impact on the test that failed. Can you check ? Thanks !!

**Comment by wszypelt** [comment]
@lgirdwood Sure, I ran the tests again, the results should be back within an hour

**Pull request merged**
